File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/PlayerActivity.java
package com.theoplayer.contentprotectionintegration;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.theoplayer.android.api.THEOplayerView;
import com.theoplayer.android.api.source.SourceDescription;

public class PlayerActivity extends AppCompatActivity {

    public final static String EXTRA_SOURCENAME = "sourceName";

    private THEOplayerView playerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player);

        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            String sourceName = extras.getString(EXTRA_SOURCENAME);
            playerView = findViewById(R.id.theoplayer);
            SourceDescription sourceDescription = SourceManager.getInstance(this).getSource(sourceName);
            playerView.getPlayer().setSource(sourceDescription);
            playerView.getPlayer().play();
        }
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (playerView != null) {
            playerView.onPause();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (playerView != null) {
            playerView.onResume();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (playerView != null) {
            playerView.onDestroy();
        }
    }
}

File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/PlayerActivity.kt
package com.theoplayer.contentprotectionintegration

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.theoplayer.android.api.THEOplayerView
import com.theoplayer.android.api.event.player.PlayerEventTypes
import com.theoplayer.contentprotectionintegration.databinding.ActivityPlayerBinding

class PlayerActivity : AppCompatActivity() {
    private lateinit var playerView: THEOplayerView
    private lateinit var viewBinding: ActivityPlayerBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.TheoTheme_Base)
        super.onCreate(savedInstanceState)

        viewBinding = ActivityPlayerBinding.inflate(layoutInflater)
        val view = viewBinding.root
        setContentView(view)

        playerView = viewBinding.theoplayer
        playerView.player.addEventListener(PlayerEventTypes.ERROR) { error ->
            viewBinding.message.text = error.errorObject.message
        }

        // Get source description from the extras bundle
        val extras = intent.extras
        if (extras != null) {
            val sourceName = extras.getString(EXTRA_SOURCE_NAME)
            val sourceDescription = SourceManager.getInstance(this).getSource(sourceName)

            playerView.player.apply {
                source = sourceDescription
                play()
            }
        }
    }

    override fun onPause() {
        super.onPause()
        playerView.onPause()
    }

    override fun onResume() {
        super.onResume()
        playerView.onResume()
    }

    override fun onDestroy() {
        super.onDestroy()
        playerView.onDestroy()
    }
}